src/java/org/apache/sqoop/avro/AvroUtil.java

```diff
@@ -114,11 +114,20 @@ public static String toAvroColumn(String column) {
    * Format candidate to avro specifics
    */
   public static String toAvroIdentifier(String candidate) {
-    String formattedCandidate = candidate.replaceAll("\\W+", "_");
-    if (formattedCandidate.substring(0,1).matches("[a-zA-Z_]")) {
-      return formattedCandidate;
+    char[] data = candidate.toCharArray();
```

Comment:
I'd suggest you create a private function `replaceWhitespace` and move your code logic there. So it will be
```
  String formattedCandidate = replaceWhitespace(candidate);
```
Second how can you make sure the string data has a valid length, so that `data[0]` is valid?